c7n/resources/vpc.py

```diff
@@ -241,6 +241,80 @@ def process(self, resources, event=None):
         return results
 
 
+@Vpc.filter_registry.register('dhcp-options')
+class DhcpOptionsFilter(Filter):
+    """Filter VPCs based on their dhcp options
+
+    """
+    schema = type_schema(
+        'dhcp-options',
+        domainname={'type': 'array', 'items': {'type': 'string'}},
+        nameserver={'type': 'array', 'items': {'type': 'string'}},
+        ntpserver={'type': 'array', 'items': {'type': 'string'}})
+    permissions = ('ec2:DescribeDhcpOptions',)
+
+    def validate(self):
+        if not any(p in self.data for p in (
+                'domainname', 'nameserver', 'ntpserver')):
+            raise FilterValidationError(
+                'at least one parameter must be provided')
+
+        if ('domainname' in self.data and
+                not isinstance(self.data.get('domainname'), list)):
+            raise FilterValidationError('domainname must be a list')
+        if ('nameserver' in self.data and
+                not isinstance(self.data.get('nameserver'), list)):
+            raise FilterValidationError('nameserver must be a list')
+        if ('ntpserver' in self.data and
+                not isinstance(self.data.get('ntpserver'), list)):
+            raise FilterValidationError('ntpserver must be a list')
+
+    def process(self, resources, event=None):
+        dmn = self.data.get('domainname', [])
+        dns = self.data.get('nameserver', [])
+        ntp = self.data.get('ntpserver', [])
+        client = local_session(self.manager.session_factory).client('ec2')
+        results = []
+        for r in resources:
+            options = client.describe_dhcp_options(
```

Comment:
i don't think its worthwhile given the low cardinality of vpcs, but dhcps options maybe common across vpcs, also more efficient to bulk query for them, ie. make mapping of dhcp-id -> vpc_ids and dhcp_id -> options.